packages/fields/src/fields/pattern-title/view.tsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Icon, lockSmall } from '@wordpress/icons';
+import { Tooltip } from '@wordpress/components';
+// @ts-ignore
+import { privateApis as patternPrivateApis } from '@wordpress/patterns';
+
+/**
+ * Internal dependencies
+ */
+import type { CommonPost } from '../../types';
+import { BaseTitleView } from '../title/view';
+import { unlock } from '../../lock-unlock';
+
+export const { PATTERN_TYPES } = unlock( patternPrivateApis );
+
+export default function PatternTitleView( { item }: { item: CommonPost } ) {
+	return (
+		<BaseTitleView item={ item } className="fields-pattern-title-field">
+			{ item.type === PATTERN_TYPES.theme && (
+				<Tooltip
+					placement="top"
+					text={ __( 'This pattern cannot be edited.' ) }
+				>
+					<Icon
+						className="fields-title-field__pattern-lock-icon"
```

Comment:
This is another case where we could improve our classnames. For example, extending the "parent" class by doing `fields-pattern-title-field__lock-icon` or similar. [Related](https://github.com/WordPress/gutenberg/pull/67449/files#r1869343274).